code/model/Translatable.php

```diff
@@ -612,13 +612,19 @@ function augmentSQL(SQLQuery &$query, DataQuery &$dataQuery = null) {
 			// the query contains this table
 			// @todo Isn't this always the case?!
 			&& array_search($baseTable, array_keys($query->getFrom())) !== false 
-			// or we're already filtering by Lang (either from an earlier augmentSQL() 
-			// call or through custom SQL filters)
-			&& !preg_match('/("|\'|`)Locale("|\'|`)/', implode(' ', $query->getWhere()))
 			//&& !$query->filtersOnFK()
 		)  {
-			$qry = sprintf('"%s"."Locale" = \'%s\'', $baseTable, Convert::raw2sql($locale));
-			$query->addWhere($qry); 
+			// Or we're already filtering by Lang (either from an earlier augmentSQL() 
+			// call or through custom SQL filters)
+			$filtersOnLocale = array_filter($query->getWhere(), function($predicates) {
+				foreach($predicates as $predicate => $params) {
```

Comment:
I think fro 4.0 compat there's no point supporting the old syntax; The augmentSQL signature will take a SQLSelect object at this point and couldn't work with 3.x code anyway.